demos/http/common/include/http_demo_utils.h

```diff
@@ -49,3 +52,60 @@ typedef int32_t ( * TransportConnect_t )( NetworkContext_t * pNetworkContext );
  */
 int32_t connectToServerWithBackoffRetries( TransportConnect_t connectFunction,
                                            NetworkContext_t * pNetworkContext );
+
+/**
+ * @brief Retrieve the path from the input URL.
+ *
+ * This function retrieves the location and length of the path from within the
+ * input the URL. The query is not included in the length returned.
+ *
+ * The URL MUST start with "http://" or "https://" to find the path.
+ *
+ * For example, if the URL is:
+ * pUrl = "https://www.somewebsite.com/path/to/item.txt?optionalquery=stuff"
+ *
+ * *pPath = "/path/to/item.txt?optionalquery=stuff"
+ * *pPathLen = 17
```

Comment:
For code examples, we normally use 
```
@code{c}
    ...
@endcode
```

Reply:
I think it was more an example of an input/output case rather than a code example, but I tried to make the comments a little clearer.